### ISSUE TITLE
Invoke switch camera method two times to switch camera from Front to Back camera

### ISSUE THREAD

**Issue by john-lq** [twilio/video-quickstart-android#548]
Hello Twilio Team,

I'm using the Twilio Video version 5.9.0.

When I switch camera from Front to Back camera I have to call `VideoCapturer.switchCamera()` method two times. It occurred on our devices(Samsung Galaxy S9(rooted device), Samsung Galaxy A50s(non-rooted device)).

I have debugged the app and see that the camera doesn't switch for the first call `VideoCapturer.switchCamera()` method. But the `cameraSource` variable is updated.
<img width="871" alt="Screen Shot 2020-07-23 at 16 00 07" src="https://user-images.githubusercontent.com/46557053/88270744-5223e680-cd00-11ea-90ad-dea6cbd1daf5.png">

Another thing: In the `CameraCapturer.java`, the `onCameraSwitchDone()` callback should be written as below.
```
               @Override
                public void onCameraSwitchDone(boolean isFrontCamera) {
                    synchronized (CameraCapturer.this) {
                        cameraSource = isFrontCamera  ? CameraSource.FRONT_CAMERA  : CameraSource.BACK_CAMERA;
                    }
                    if (listener != null) {
                        listener.onCameraSwitched();
                    }
                }
```
Could you help me to check that!

Thanks,
John

**Comment by aaalaniz** [comment]
Hey @john-lq 

Nice find with the camera switch logic. 👍 

I filed a bug and we will fix this in an upcoming release.

As for the switch camera behavior, I suspect the issue is related to the device containing more than 2 cameras (front and back). Unlike Camera2Capturer, CameraCapturer cycles through the cameras in order when calling `switchCamera`. As a result, in cases where a device contains for example 4 cameras, the switchCamera logic will just go to the next camera which is not guaranteed to be oriented on a different side of the device than the previous camera. Your workaround (calling switch camera multiple times) is recommended in these cases where CameraCapturer is required. However, if Camera2Capturer is supported on these devices, then you have the option to call `switchCamera` with a specific camera ID so you do not have to iterate through cameras.

Thanks!

**Comment by john-lq** [comment]
Hey @aaalaniz ,

I tried to use Camera2Capture. It works well on both devices.

Thanks,
John

**Comment by aaalaniz** [comment]
Hey @john-lq 

This issue was resolved in 5.10.2.

Thank you for finding and helping resolve the issue!
